File: I1_MVPAndRetrofit/app/src/main/java/com/example/tsvetelinastoyanova/weatherreportrevisited/util/CityEntityAdapter.java
package com.example.tsvetelinastoyanova.weatherreportrevisited.util;

import com.example.tsvetelinastoyanova.weatherreportrevisited.City;
import com.example.tsvetelinastoyanova.weatherreportrevisited.data.CityEntity;
import com.example.tsvetelinastoyanova.weatherreportrevisited.model.WeatherObject;

import java.util.ArrayList;
import java.util.List;

public class CityEntityAdapter {

    public static List<City> convertCityEnitiesToCities(List<CityEntity> cityEntities) {
        List<City> cities = new ArrayList<>();
        for (CityEntity cityEntity : cityEntities) {
            cities.add(new City(cityEntity.getName(), cityEntity.getLastTemperature(), cityEntity.getLastImageId()));
        }
        return cities;
    }

    public static City convertCityEntityToCity(CityEntity cityEntity) {
        return new City(cityEntity.getName(), cityEntity.getLastTemperature(), cityEntity.getLastImageId());
    }

    public static CityEntity convertWeatherObjectToCityEntity(WeatherObject object) {
        CityEntity c = new CityEntity();
        c.setName(object.getName());
        c.setCityId(object.getId());
        c.setLastTemperature(object.getMain().getTemp());
        String imageId = object.getWeather().get(0).getIcon();
        c.setLastImageId(ImageOperator.getImageIdFromString(imageId));
        return c;
    }

    public static City convertWeatherObjectToCity(WeatherObject weatherObject) {
        String image = weatherObject.getWeather().get(0).getIcon();
        return new City(weatherObject.getName(), weatherObject.getMain().getTemp(), ImageOperator.getImageIdFromString(image));
    }
}

File: I1_MVPAndRetrofit_Kotlin/app/src/main/java/com/example/tsvetelinastoyanova/weatherapp/util/CityEntityAdapter.kt
package com.example.tsvetelinastoyanova.weatherapp.util

import com.example.tsvetelinastoyanova.weatherapp.City
import com.example.tsvetelinastoyanova.weatherapp.data.CityEntity
import com.example.tsvetelinastoyanova.weatherapp.model.currentweather.CurrentWeatherObject
import com.example.tsvetelinastoyanova.weatherapp.model.forecast.ForecastObject
import java.util.ArrayList

object CityEntityAdapter {

    fun convertCityEnitiesToCities(cityEntities: List<CityEntity>): List<City> {
        val cities = ArrayList<City>()
        for (cityEntity in cityEntities) {
            cities.add(City(cityEntity.name, cityEntity.lastTemperature, cityEntity.lastImageId))
        }
        return cities
    }

    fun convertCityEntityToCity(cityEntity: CityEntity): City {
        return City(cityEntity.name, cityEntity.lastTemperature, cityEntity.lastImageId)
    }


    fun convertWeatherObjectToCityEntity(weatherObjectsList: CurrentWeatherObject): CityEntity {
        val c = CityEntity()
        c.name = weatherObjectsList.name
        c.cityId = weatherObjectsList.id
        c.lastTemperature = weatherObjectsList.main.temp
        val imageId = weatherObjectsList.weather[0].icon
        c.lastImageId = ImageOperator.getImageIdFromString(imageId)
        return c
    }

    fun convertFirstWeatherObjectToCityEntity(weatherObjectsList: ForecastObject): CityEntity {
        val c = CityEntity()
        c.name = weatherObjectsList.city.name
        c.cityId = weatherObjectsList.city.id
        c.lastTemperature = weatherObjectsList.list[0].main.temp
        val imageId = weatherObjectsList.list[0].weather[0].icon
        c.lastImageId = ImageOperator.getImageIdFromString(imageId)
        return c
    }

    fun convertWeatherObjectToCity(weatherObject: ForecastObject): City {
        val image = weatherObject.list[0].weather[0].icon
        return City(weatherObject.city.name, weatherObject.list[0].main.temp, ImageOperator.getImageIdFromString(image))
    }

    fun reduceWeatherObjectsList(weatherObjectsList: ForecastObject): ForecastObject {
        return ForecastObject(cod = weatherObjectsList.cod, message = weatherObjectsList.message,
                city = weatherObjectsList.city, cnt = weatherObjectsList.cnt, list = weatherObjectsList.list.subList(0, 10))
    }
}
